rotkehlchen/globaldb/assets_management.py

```diff
@@ -0,0 +1,125 @@
+import json
+import logging
+import tempfile
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Optional, Union
+from zipfile import ZipFile
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.assets.typing import AssetData, AssetType
+from rotkehlchen.errors import InputError, UnknownAsset
+from rotkehlchen.globaldb.handler import GLOBAL_DB_VERSION, GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.validation.validation_schemas import ExportedAssetsSchema
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def import_assets_from_file(
+    path: Path,
+    msg_aggregator: 'MessagesAggregator',
+    db_handler: 'DBHandler',
+) -> None:
+    """
+    Import assets from the file at the defined path.
+    This function can raise:
+    - ValidationError: If the format of the file is not correct
+    - InputError: If the version of the file is not valid for the current
+    globaldb version
+    """
+    globaldb = GlobalDBHandler()
+    with open(path) as f:
+        data = ExportedAssetsSchema().loads(f.read())
+
+    # First assert that the version is correct
+    if int(data['version']) != GLOBAL_DB_VERSION:
+        raise InputError('Provided file is for a different version of rotki')
+
+    identifiers = []
+    for asset_data in data['assets']:
+        # Check if we already have the asset with that name and symbol. It is possible that
+        # we have added a missing asset
+        asset_type = asset_data['asset_type']
+        asset_ref: Union[Optional[List[str]], Optional[AssetData]]
+        if asset_type == AssetType.ETHEREUM_TOKEN:
+            asset_ref = globaldb.get_asset_data(
+                identifier=asset_data['identifier'],
+                form_with_incomplete_data=True,
+            )
+        else:
+            asset_ref = globaldb.check_asset_exists(
+                asset_type=asset_type,
+                name=asset_data['name'],
+                symbol=asset_data['symbol'],
+            )
+        if asset_ref is not None:
+            msg_aggregator.add_warning(
+                f'Tried to import existing asset {asset_data["identifier"]} with '
+                f'name {asset_data["name"]}',
+            )
+            continue
+
+        try:
+            globaldb.add_asset(
+                asset_id=asset_data['identifier'],
+                asset_type=asset_type,
+                data=asset_data['extra_information'],
+            )
+        except InputError as e:
+            log.error(
+                f'Failed to import asset with {asset_data["identifier"]=}',
+                f'{asset_type=} and {asset_data=}. {str(e)}',
+            )
+            msg_aggregator.add_warning(
```

Comment:
```suggestion
            msg_aggregator.add_error(
```

If you log an error also use msg_aggregator with error